virtual_machine.go

```diff
@@ -361,7 +361,9 @@ func (v *VirtualMachine) deleteCloudInitISO(ctx context.Context) (ok bool, err e
 func (v *VirtualMachine) Migrate(ctx context.Context, target, targetstorage string) (task *Task, err error) {
 	var upid UPID
 	params := map[string]string{
-		"target": target,
+		"target":           target,
+		"online":           "1",
```

Comment:
should we consider a live and offline migration function, i feel like this forces live migrations all the time?

Reply:
I wouldn't really say it forces anything, as it doesn't change much aside from changing an error into a working migration.

Default behaviour:

* offline VM are migrated
* online VM return a 500 error asking to 

With those flags set:

* offline VM are still migrated offline
* online VM are now migrated

That being said, if your idea is to keep control over optional parameters rather than specifically splitting the calls between cold and hot, I think just moving every optional parameter to a `VirtualMachineMigrateOptions` struct would probably be a better idea, as there is some more not being addressed here (like `bwlimit`, `migration_type`, `migration_network`, etc) and aim for a similar design as the `Clone` method.

Reply:
i agree with you, options is better... you want to switch this PR to that method?

Reply:
Reworked PR to follow the Clone model